server/src/api/shoutoutLogFilter.ts

```diff
@@ -0,0 +1,25 @@
+import api from '../shared/api';
+import { ShoutoutLog } from '../types/ShoutoutLog';
+import { User } from '../types/User';
+
+interface ShoutoutLogFilterOptions {
+  count: number;
+  usernameFrom?: User['username'];
+  usernameTo: User['username'];
+}
+
+const shoutoutLogFilter = (options: ShoutoutLogFilterOptions) => {
+  const filteredOptions = (
+    Object.keys(options) as (keyof ShoutoutLogFilterOptions)[]
+  ).reduce(
+    (accumulator, key) =>
+      options[key] ? { ...accumulator, [key]: options[key] } : accumulator,
+    {}
+  );
+  const searchParams = new URLSearchParams(filteredOptions);
```

Comment:
`URLSearchParams` also accepts an array of arrays,
it'd be cleaner to rather do the above "cleanup" using arrays:

```suggestion
const filteredOptions = Object.keys(options)
  .map((key) => ([key, options[key]))
  .filter(([, value]) => value !== undefined);

const searchParams = new URLSearchParams(filteredOptions);
```